frontend/src/metabase/common/components/FilterPicker/FilterPicker.unit.spec.tsx

```diff
@@ -425,34 +424,87 @@ describe("FilterPicker", () => {
     }
 
     it("should create a filter with a custom expression", async () => {
-      const { onSelect, onSelectLegacy } = setup();
+      const { query, getNextFilter, onSelect } = setup();
 
       userEvent.click(screen.getByText(/Custom expression/i));
       await editExpressionAndSubmit("[[Total] > [[Discount]{enter}");
 
       await waitFor(() =>
-        expect(onSelectLegacy).toHaveBeenCalledWith([
-          ">",
-          ["field", ORDERS.TOTAL, null],
-          ["field", ORDERS.DISCOUNT, null],
-        ]),
+        expect(onSelect).toHaveBeenCalledWith(expect.anything()),
       );
-      expect(onSelect).not.toHaveBeenCalled();
+
+      const filter = getNextFilter();
+      const { args, operator, options } = Lib.expressionParts(query, 0, filter);
+      const [left, right] = args;
+      const leftMetadata = checkNotNull(left) as Lib.ColumnMetadata;
+      const rightMetadata = checkNotNull(right) as Lib.ColumnMetadata;
```

Comment:
We could introduce type guards to avoid these casts, WDYT?

Reply:
You're not supposed to be calling these functions in a test, it's an MBQL implementation detail. `expressionParts` shouldn't be exported from the library

Reply:
Calling something simple, like `displayInfo` is enough

Reply:
a25f2e006cfbcb6e1e9d0ce336d399a0d7e27811